pkg/config/values.go

```diff
@@ -6,27 +6,19 @@ package config
 
 import (
 	"github.com/hashicorp/hcl/v2"
-	"github.com/hashicorp/hcl/v2/hclparse"
 	"github.com/zclconf/go-cty/cty"
 )
 
-// LoadValuesFile reads the file at the given path and parses it as a
+// LoadValues reads the list of hcl.File and parses it as a
 // "values file" (flat key.value HCL config) for later use in the
 // `EvalContext`.
 //
 // Adapted from
 // https://github.com/hashicorp/terraform/blob/d4ac68423c4998279f33404db46809d27a5c2362/configs/parser_values.go#L8-L23
-func LoadValuesFile(path string) (map[string]cty.Value, hcl.Diagnostics) {
-	hclParser := hclparse.NewParser()
-	varsFile, diags := hclParser.ParseHCLFile(path)
-	if diags != nil {
-		return nil, diags
-	}
+func LoadValues(files []*hcl.File) (map[string]cty.Value, hcl.Diagnostics) {
```

Comment:
`LoadValues` is a very vague name IMO. I had to read the entire implementation to understand what this function does.

Could you rename this function so that the reader could get a hint regarding what it does when seeing its name?

To break this down, "load" is a vague verb and "values" is a vague noun. I would replace "load" with a more concrete action and "values" with a more concrete entity.

Idea (not sure if a good one): name the function `hclVarsFromFile()`, modify it to accept a single file and call it for every relevant file.